### ISSUE TITLE
Directing layout based on node classes.

### ISSUE THREAD

**Issue by Tadasu85** [cytoscape/cytoscape.js#1661]
Hello I am the dev working on steemstars using cytoscape and I am wondering if there is a way to specify the edge distance or the bounding based on the class of the node. Also thank you guys for adding my app to the cytoscape js homepage if you could also add the website itself that would be great, it resides here: https://steemstars.herokuapp.com/ Thank you guys again!

**Comment by maxkfranz** [comment]
There are two approaches for per-element layout specification:

(1) Run a layout only on a subset of elements.

```js
cy.$('.foo').layout( /* ... */ );
```

(2) Specify functions that return per-element values for layout options.

```js
cy.layout({
  name: 'bar',
  someOption: function( ele ){ return ele.hasClass('foo') ? 1 : 0; }
});
```

**Comment by Tadasu85** [comment]
beautiful thank you.

**Comment by Tadasu85** [comment]
Alright I have been playing with this for some time, and I have found that the only layout option that actually works with either of these implementations is random ALL of the rest of the layout algos that I have attempted using end up breaking and just displaying as grids. Is there a layout that maybe just considers node weight and sets edge length based on that? I have played with some of the supposed physics layouts but none of them seem to consider node weight...or if they do so they seem to do so in such a minuscule way that it is useless.
